adapters/repos/db/lsmkv/segment_replace_strategy.go

```diff
@@ -28,22 +28,30 @@ func (s *segment) get(key []byte) ([]byte, error) {
 
 	before := time.Now()
 
-	if !s.bloomFilter.Test(key) {
-		s.bloomFilterMetrics.trueNegative(before)
+	if s.bloomFilter != nil && !s.bloomFilter.Test(key) {
```

Comment:
Same here.  Testing against useBloomFilter will ensure that the behaviour is consistent with the settings.